src/components/tree-views-container.js

```diff
@@ -500,64 +501,67 @@ const TreeViewsContainer = () => {
+                    const isRootDirectory =
+                        eventData.headers['isRootDirectory'];
+                    const directoryUuid = eventData.headers['directoryUuid'];
+                    const error = eventData.headers['error'];
+                    const elementName = eventData.headers['elementName'];
+
+                    displayErrorIfExist(error, elementName);
+
+                    if (isRootDirectory) {
+                        updateRootDirectories();
+                        if (
+                            notificationTypeHeader ===
+                                notificationType.DELETE_DIRECTORY &&
+                            selectedDirectoryRef.current === directoryUuid
+                        ) {
+                            dispatch(setSelectedDirectory(null));
+                        }
+                        return;
                     }
-                    return;
-                }
 
-                if (directoryUuid) {
-                    if (directoryUuid === selectedDirectoryRef.current) {
-                        updateDirectoryTreeAndContent(directoryUuid);
-                    } else {
-                        updateDirectoryTree(directoryUuid);
+                    if (directoryUuid) {
+                        if (directoryUuid === selectedDirectoryRef.current) {
+                            updateDirectoryTreeAndContent(directoryUuid);
+                        } else {
+                            updateDirectoryTree(directoryUuid);
+                        }
                     }
                 }
-            }
-        };
-        ws.onclose = function () {
-            if (!websocketExpectedCloseRef.current) {
-                console.error('Unexpected Notification WebSocket closed');
-            }
-        };
-        ws.onerror = function (event) {
-            console.error('Unexpected Notification WebSocket error', event);
-        };
-        return ws;
+            };
+            ws.onclose = function () {
+                if (!websocketExpectedCloseRef.current) {
+                    console.error('Unexpected Notification WebSocket closed');
+                }
+            };
+            ws.onerror = function (event) {
+                websocketRef.current = null;
+                websocketRef.current.close();
+                console.error('Unexpected Notification WebSocket error', event);
+            };
+        }
+        return websocketRef.current;
     }, [
+        dispatch,
         displayErrorIfExist,
-        updateDirectoryTreeAndContent,
         updateDirectoryTree,
+        updateDirectoryTreeAndContent,
         updateRootDirectories,
-        dispatch,
     ]);
 
     useEffect(() => {
-        const ws = connectNotificationsUpdateStudies();
-        // Note: dispatch doesn't change
-        // cleanup at unmount event
-        return function () {
-            ws.close();
-        };
+        connectNotificationsUpdateStudies();
```

Comment:
We never close the websocket then ?
We should have just a useEffect with empty dependency array parameter then create the ws at mounting and closing it at unmounting (return function).
The ws creation will be called only once without using useRef.
The problem comes when you start to add dependencies in the UseEffect then the behavior change completly.

Reply:
Modifications done

Reply:
Tests have been done and the websocket is properly recreated when leaving/reopening the navigator, and also when gateway service is stopped and restarted.